enhancements/installer/telemeter-metrics.md

```diff
@@ -0,0 +1,430 @@
+* `le` - the standard Prometheus label for histogram buckets, this represents how long a command took complete. The current proposal is to have a 5 minute step starting at 15 minutes and maxing at 60.
+
+##### cluster_installation_waitfor
+UPI installs utilize the wait-for command, which would be tracked with a similar histogram:
+```
+# HELP cluster_installation_waitfor represents a single run of wait-for by the OpenShift installer.
+# TYPE cluster_installation_waitfor histogram
+cluster_installation_waitfor_bucket{target="cluster-complete",platform="aws",result="success",version="4.2",os="linux",le="5"} 0
+cluster_installation_waitfor_bucket{target="cluster-complete",platform="aws",result="success",version="4.2",os="linux",le="10"} 0
+cluster_installation_waitfor_bucket{target="cluster-complete",platform="aws",result="success",version="4.2",os="linux",le="15"} 1
+cluster_installation_waitfor_bucket{target="cluster-complete",platform="aws",result="success",version="4.2",os="linux",le="20"} 1
+cluster_installation_waitfor_bucket{target="cluster-complete",platform="aws",result="success",version="4.2",os="linux",le="25"} 1
+cluster_installation_waitfor_bucket{target="cluster-complete",platform="aws",result="success",version="4.2",os="linux",le="30"} 1
+cluster_installation_waitfor_bucket{target="cluster-complete",platform="aws",result="success",version="4.2",os="linux",le="+Inf"} 1
+cluster_installation_waitfor_sum{target="cluster-complete",platform="aws",result="success",version="4.2",os="linux"} 15
+cluster_installation_waitfor_count{target="cluster-complete",platform="aws",result="success",version="4.2",os="linux"} 1
+```
+
+###### Label Values
+Potential label values are the same as the previous metric, except:
+* `le` has values between 5 and 30 minutes
+* `target` and `result` are dependent:
+  - `bootstrap-complete`
+      - `BootstrapFailed`
+      - `APIFailed`
+      - `Success`
+  - `install-complete`
+      - `OperatorsDegraded`
+      - `Success`
+
+##### cluster_installation_manifests and cluster_installation_ignition
+These metrics are simple counters totalling the number of runs for `openshift-install create manifests` or `openshift-install create ignition-configs`:
+
+```
+# HELP cluster_installation_manifests represents a single run of create manifests by the OpenShift installer.
+# TYPE cluster_installation_manifests counter
+cluster_installation_manifests{platform="aws",result="success",version="4.2",os="linux"} 1
+
+# HELP cluster_installation_ignition represents a single run of create ignition-configs by the OpenShift installer.
+# TYPE cluster_installation_ignition counter
+cluster_installation_ignition{platform="aws",result="success",version="4.2",os="linux"} 1
+```
+
+Possible `result` values are `success` and are `error`, otherwise all possible label values are the same as `cluster_installation_create`. The sample value would always be one in order to aggregate to a total count of runs of this command.
```

Comment:
Some of your metrics use `success` and some use `Success`.  You should probably pick one way for all the metrics.

Reply:
Also, `success` and `error` are hard to extend later.  For example, you could sum `result="success"` over the whole to get a success rate.  But if you sum `result="error"` over the whole to get an error rate, that would break down if we subsequently split `error` into sub-cases (e.g. `throttled checking credentials`).  I think it's safer to have an `error` label that we set to the empty string for success, and which we set to `other` for uncategorized errors.

Reply:
I agree with standardizing capitalization. Wouldn't you just query for `not success`?

Reply:
> Wouldn't you just query for `not success`?

That would work.  I just think `result="error"` is a foot-gun, and would rather not have it around ;).  `result="other error"` avoids the foot-gun, but feels (subjectively) awkward.  `error="none"` and `error="other"` feel good to me, but maybe they feel awkward to you?